src/main/java/com/opentok/OpenTok.java

```diff
@@ -291,6 +291,35 @@ public Session createSession() throws OpenTokException {
         return createSession(null);
     }
 
+    public void signal(String sessionId, SignalProperties props) throws OpenTokException , RequestException, InvalidArgumentException {
+
+        if (sessionId == null || sessionId.isEmpty() ) {
+            throw new InvalidArgumentException("Session string null or empty");
+        }
+        try {
+            this.client.signal(sessionId,null,props);
+
+        } catch (Exception e)
+        {
+            throw e;
+        }
+
+    }
+
+    public void signal(String sessionId, String connectionId, SignalProperties props) throws OpenTokException , RequestException , InvalidArgumentException {
```

Comment:
Apologies for re-reviewing the PR, but I just noticed that I missed this.

The `connectionId` should be third parameter IMHO because it's an optional parameter. I understand in Java you have `overloads`, but for the sake of consistency with other server SDKs we should keep it the same format.

Please see the PHP and Python SDK for more details: https://github.com/opentok/Opentok-Python-SDK/issues/127

Reply:
The connectionId is not optional. It is a different API pointing to a distinct endpoint.